### ISSUE TITLE
Header frame has CONTINUATION frames, and set END_STREAM.

### ISSUE THREAD

**Issue by CodeLife2012** [h2o/h2o#2045]
[expect_continuation_of_headers](https://github.com/h2o/h2o/blob/409064278be76b049405a864c6f08a7b81c8c1ca/lib/common/http2client.c#L374) only use the last CONTINUATION frame's END_STREAM flag.

But from the [Headers doc](https://httpwg.org/specs/rfc7540.html#HEADERS), 

![image](https://user-images.githubusercontent.com/4277861/57367524-b9b54b00-71bb-11e9-901a-7f26ad07fa42.png)

CONTINUATION frame may not have a END_STREAM flag

**Comment by CodeLife2012** [comment]
```
  GET /resource HTTP/1.1          HEADERS
  Host: example.org          ==>     + END_STREAM
  {empty data}                       - END_HEADERS
                                       :method = GET
                                       :path = /resource
                                       :scheme = https
                                       :host = example.org

                                   CONTINUATION
                                     + END_HEADERS
                                       cookie = xxx
```

Can header frame send like this?

**Comment by i110** [comment]
Thank you for reporting this issue. You're definitely right. `END_STREAM` flag is not defined in CONTINUATION frame anyway. I'll fix it

**Comment by i110** [comment]
@CodeLife2012 

> Is response END_STREAM without data frame is correct?

yes it's valid HTTP/2 response if the response contains no payload, for example, 204 response.

> Does h2o handle the Malformed Requests and Responses?

What is the point of your question?
